lib/paper_trail/queries/versions/where_object_changes_from.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module PaperTrail
+  module Queries
+    module Versions
+      # For public API documentation, see `where_object_changes_from` in
+      # `paper_trail/version_concern.rb`.
+      # @api private
+      class WhereObjectChangesFrom
+        # - version_model_class - The class that VersionConcern was mixed into.
+        # - attributes - A `Hash` of attributes and values. See the public API
+        #   documentation for details.
+        # @api private
+        def initialize(version_model_class, attributes)
+          @version_model_class = version_model_class
+          @attributes = attributes
+        end
+
+        # @api private
+        def execute
+          if PaperTrail.config.object_changes_adapter&.respond_to?(:where_object_changes_from)
+            return PaperTrail.config.object_changes_adapter.where_object_changes_from(
+              @version_model_class, @attributes
+            )
+          end
+
+          case @version_model_class.columns_hash["object_changes"].type
+          when :jsonb, :json
+            json
+          else
+            text
+          end
+        end
+
+        private
+
+        # @api private
+        def json
+          predicates = []
+          values = []
+          @attributes.each do |field, value|
+            predicates.push(
+              "(object_changes->>? ILIKE ?)"
+            )
+            values.concat([field, "[#{value.to_json},%"])
+          end
+          sql = predicates.join(" and ")
+          @version_model_class.where(sql, *values)
+        end
+
+        # @api private
+        def text
+          arel_field = @version_model_class.arel_table[:object_changes]
+
+          @attributes.map do |field, value|
+            ::PaperTrail.serializer.where_object_changes_from_condition(arel_field, field, value)
+          end
```

Comment:
This is honestly relying on `where_object_changes_from_condition` in either serializer raising an exception. I personally feel we might be better off explicitly raising here, but I also wanted to mirror some of the implementation in [WhereObjectChanges](https://github.com/paper-trail-gem/paper_trail/blob/865f069e091d69ef9c136a27f3372f9d24eb37e9/lib/paper_trail/queries/versions/where_object_changes.rb#L64-L71), though I'm not sure if that's valuable.

Reply:
I pushed another commit which includes the same full implementation as exists in `WhereObjectChanges`. The reason for that is [Custom Serializers](https://github.com/paper-trail-gem/paper_trail#6b-custom-serializer).

However, I don't think I have a great testing story to ensure this works as expected. The [custom serializer](https://github.com/paper-trail-gem/paper_trail/blob/c3c58b739a62100b83ff79762c91f3dbd857761d/spec/support/custom_json_serializer.rb) available for specs already extends the JSON Serializer, which will raise the exception (unless we modify `CustomJsonSerializer` to *not* do that.

@jaredbeck, do you have any thoughts on if there's more work we can or should do to ensure custom serializers will play nicely with this?